service/worker/scanner/workflow.go

```diff
@@ -58,9 +58,11 @@ func (s scannerCtxExecMgrFactory) NewExecutionManager(shardID int32) (persistenc
 const (
 	scannerContextKey = contextKey(0)
 
-	maxConcurrentActivityExecutionSize     = 10
-	maxConcurrentWorkflowTaskExecutionSize = 10
-	infiniteDuration                       = 20 * 365 * 24 * time.Hour
+	maxConcurrentActivityExecutionSize = 10
+	maxConcurrentWorkflowExecutionSize = 10
+	maxConcurrentActivityTaskPollers   = 16
+	maxConcurrentWorkflowTaskPollers   = 16
```

Comment:
concurrent poller won't be larger than concurrent execution. 
Jump from 2 -> 16 seems aggressive. I suggest to use 8.
Ideally, this should be dynamic config. :)